### DISCUSSION TITLE
jax.jit missing manual ?

### DISCUSSION THREAD

**Discussion by mitkotak** [jax-ml/jax#23051] [General]
I am wondering what the equivalent of [this](https://docs.google.com/document/d/1y5CRfMLdwEoF1nTk9q8qEu1mgMUuUtvhklPKJ2emLU8/edit) is in JAX-land ?

I am a big fan of JAX docs (particularly autodidax) but I am wondering if there's a similar way to reason through the entire stack all the way down to the Triton kernels that XLA is generating ?

**Comment by yashk2810** [comment]
JAX has `jax.jit` for automatic partitioning: https://jax.readthedocs.io/en/latest/notebooks/Distributed_arrays_and_automatic_parallelization.html

and `shard_map` for manual partitioning: https://jax.readthedocs.io/en/latest/notebooks/shard_map.html

If you want to write kernels, check out pallas: https://jax.readthedocs.io/en/latest/pallas/quickstart.html
